ros/src/tl_detector/tl_detector.py

```diff
@@ -219,33 +212,30 @@ def get_next_tl_waypoint_index(self, stop_line_positions):
                 self.car_direction = -1
         else:
             self.car_direction = 1
+
         self.last_car_wp_idx = car_wp_idx
 
         # Find the closest upcoming stop line waypoint index in front of the car
         INDEX_DISTANCE_THRESHOLD = 150
-        min_dist = 1e+10
-        wp1 = self.base_waypoints_np[car_wp_idx]
-        for i in range(car_wp_idx, car_wp_idx+self.car_direction*INDEX_DISTANCE_THRESHOLD, self.car_direction):
-            if np.isin(i, self.stop_lines_wp_idxs):
-                wp2 = self.base_waypoints_np[i]
-                dist = np.linalg.norm(wp1 - wp2)
-                if dist < min_dist:
-                    tl_wp_idx = i
-                    min_dist = dist
 
+        light_dist = [0 < (self.car_direction *(line_index - car_wp_idx)) < INDEX_DISTANCE_THRESHOLD
+                        for line_index in self.stop_lines_wp_idxs]
+        
         self.last_in_range = self.in_range
-        if min_dist < 1e+10:
+
+        if any(light_dist):
```

Comment:
thanks :) I know 1e+10 looked ugly but was in a rush to finish optimising on saturday